internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -1099,32 +1168,91 @@ func TestMachineSetReconciler_syncMachines(t *testing.T) {
+	g.Expect(updatedInfraMachine.GetManagedFields()).ShouldNot(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:metadata", "f:labels"}))
+	g.Expect(updatedInfraMachine.GetManagedFields()).ShouldNot(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:metadata", "f:annotations"}))
+	g.Expect(updatedInfraMachine.GetManagedFields()).Should(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:spec"}))
+
+	// The BootstrapConfig should have ownership of "labels" and "annotations" transferred to
+	//	// "capi-machineset" manager.
+	updatedBootstrapConfig := bootstrapConfig.DeepCopy()
+	g.Expect(env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(updatedBootstrapConfig), updatedBootstrapConfig))
+
+	// Verify ManagedFields
+	g.Expect(updatedBootstrapConfig.GetManagedFields()).Should(
+		ssa.MatchFieldOwnership(machineSetManagerName, metav1.ManagedFieldsOperationApply, contract.Path{"f:metadata", "f:labels"}))
+	g.Expect(updatedBootstrapConfig.GetManagedFields()).Should(
+		ssa.MatchFieldOwnership(machineSetManagerName, metav1.ManagedFieldsOperationApply, contract.Path{"f:metadata", "f:annotations"}))
+	g.Expect(updatedBootstrapConfig.GetManagedFields()).ShouldNot(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:metadata", "f:labels"}))
+	g.Expect(updatedBootstrapConfig.GetManagedFields()).ShouldNot(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:metadata", "f:annotations"}))
+	g.Expect(updatedBootstrapConfig.GetManagedFields()).Should(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:spec"}))
+
+	//
+	// Verify In-place mutating fields
+	//
+
+	// Update the MachineSet and verify the in-mutating fields are propagated.
+	ms.Spec.Template.Labels = map[string]string{
+		"preserved-label": "preserved-value",
+		"modified-label":  "modified-value-2",
+	}
+	ms.Spec.Template.Annotations = map[string]string{
+		"preserved-annotation": "preserved-value",
+		"modified-annotation":  "modified-value-2",
+	}
+	ms.Spec.Template.Spec.NodeDrainTimeout = duration10s
+	ms.Spec.Template.Spec.NodeDeletionTimeout = duration10s
+	ms.Spec.Template.Spec.NodeVolumeDetachTimeout = duration10s
+	g.Expect(reconciler.syncMachines(ctx, ms, []*clusterv1.Machine{updatedInPlaceMutatingMachine, deletingMachine})).To(Succeed())
+
+	// Verify in-place mutable fields are updated on the Machine.
+	updatedInPlaceMutatingMachine = inPlaceMutatingMachine.DeepCopy()
+	g.Expect(env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(updatedInPlaceMutatingMachine), updatedInPlaceMutatingMachine))
 	// Verify Labels
-	g.Expect(updatedInPlaceMutatingMachine.Labels).Should(HaveKeyWithValue("machine-set-matching-label", "true"))
```

Comment:
nit: can we change all those label/annotation validations by ones which check the exact content of the map?
```go
	g.Expect(updatedInPlaceMutatingMachine.Labels).Should(Equal(map[string]string{
	    ....	
	}))
```

I know we have to add a few additional labels, but I think it's better as it ensures that the map looks 100% exactly like we expect it too.